ice-server/src/IceServer.cpp

```diff
@@ -44,63 +48,61 @@ IceServer::IceServer(int argc, char* argv[]) :
+    } else {
+        qDebug() << "Packet version mismatch for packet" << packet.getType()
+            << "(" << nameForPacketType(packet.getType()) << ") from" << packet.getSenderSockAddr();
         
-        auto packet = Packet::fromReceivedPacket(std::move(buffer), packetSizeWithHeader, sendingSockAddr);
-
-        PacketType::Value packetType = packet->getType();
-
-        if (packetType == PacketType::ICEServerHeartbeat) {
-            SharedNetworkPeer peer = addOrUpdateHeartbeatingPeer(*packet);
-
-            // so that we can send packets to the heartbeating peer when we need, we need to activate a socket now
-            peer->activateMatchingOrNewSymmetricSocket(sendingSockAddr);
-        } else if (packetType == PacketType::ICEServerQuery) {
-            QDataStream heartbeatStream(packet.get());
-
-            // this is a node hoping to connect to a heartbeating peer - do we have the heartbeating peer?
-            QUuid senderUUID;
-            heartbeatStream >> senderUUID;
-
-            // pull the public and private sock addrs for this peer
-            HifiSockAddr publicSocket, localSocket;
-            heartbeatStream >> publicSocket >> localSocket;
+        return false;
+    }
+}
 
-            // check if this node also included a UUID that they would like to connect to
-            QUuid connectRequestID;
-            heartbeatStream >> connectRequestID;
+void IceServer::processPacket(std::unique_ptr<udt::Packet> packet) {
+    PacketType::Value packetType = packet->getType();
+    
+    if (packetType == PacketType::ICEServerHeartbeat) {
+        SharedNetworkPeer peer = addOrUpdateHeartbeatingPeer(*packet);
+        
+        // so that we can send packets to the heartbeating peer when we need, we need to activate a socket now
+        peer->activateMatchingOrNewSymmetricSocket(packet->getSenderSockAddr());
+    } else if (packetType == PacketType::ICEServerQuery) {
+        QDataStream heartbeatStream(packet.get());
+        
+        // this is a node hoping to connect to a heartbeating peer - do we have the heartbeating peer?
+        QUuid senderUUID;
+        heartbeatStream >> senderUUID;
+        
+        // pull the public and private sock addrs for this peer
+        HifiSockAddr publicSocket, localSocket;
+        heartbeatStream >> publicSocket >> localSocket;
+        
+        // check if this node also included a UUID that they would like to connect to
+        QUuid connectRequestID;
+        heartbeatStream >> connectRequestID;
+        
+        SharedNetworkPeer matchingPeer = _activePeers.value(connectRequestID);
+        
+        if (matchingPeer) {
+            
+            qDebug() << "Sending information for peer" << connectRequestID << "to peer" << senderUUID;
+            
+            // we have the peer they want to connect to - send them pack the information for that peer
+            sendPeerInformationPacket(*(matchingPeer.data()), &packet->getSenderSockAddr());
```

Comment:
I know you just reindented that but can you replace that `*(matchingPeer.data())` with `*matchingPeer`?